sys/hashes/sha1.c

```diff
@@ -183,7 +183,7 @@ void sha1_init_hmac(sha1_context *s, const unsigned char *key, size_t key_length
         while(key_length--) {
             sha1_update_byte(s, (unsigned char) *key++);
         }
-        memcpy(s->key_buffer, sha1_final(s), SHA1_DIGEST_LENGTH);
+        sha1_final(s, s->key_buffer);
```

Comment:
I think I am leaving it at that. The unification is not perfect (sha256 still uses unsigned char instead of uint8_t in some places, the hmac interface is missing some functions in sha256, md5 cannot calculate a hmac at all), but for the time beeing that should do.

In the long run, a real common interface for the digests should be implemented, like it was done for the ciphers.